[BUG] HC32 + ProUI `__iCliRetVal / __iSeiRetVal not declared in this scope`

### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

paging @shadow578

These two functions not found:
```r
HAL/HC32/HAL.h:122:9: error: '__iCliRetVal' was not declared in this scope
HAL/HC32/HAL.h:126:9: error: '__iSeiRetVal' was not declared in this scope
```

**Marlin\src\HAL\HC32\HAL.h**
```y
#define CRITICAL_SECTION_START        \
  uint32_t primask = __get_PRIMASK(); \
  (void)__iCliRetVal()

#define CRITICAL_SECTION_END \
  if (!primask)              \
  (void)__iSeiRetVal()
```


so apparently `__iSeiRetVal` and `__iCliRetVal` are found in **.platformio\packages\toolchain-atmelavr\avr\include\util\atomic.h**

I noticed in **Marlin\src\HAL\STM32F1\HAL.h** it has `#include <util/atomic.h>` which HC32 does not.

The issue resides here, in **libs\buzzer.cpp**, in function `Buzzer::tick()`:
```y
      #if ENABLED(EXTENSIBLE_UI) && DISABLED(EXTUI_LOCAL_BEEPER)
        CRITICAL_SECTION_START();
        ExtUI::onPlayTone(state.tone.frequency, state.tone.duration);
        CRITICAL_SECTION_END();
```

Since ProUI is now an `EXTENSIBLE_UI`, this get's enabled as well, otherwise it would not be an issue.

A workaround is to simply omit it if `defined(ARDUINO_ARCH_HC32)`:
e.g.:
```y
#if ENABLED(EXTENSIBLE_UI) && NONE(EXTUI_LOCAL_BEEPER, ARDUINO_ARCH_HC32)
```
ProUI doesn't have a useable `onPlayTone()`  
Otherwise somehow make the **HC32/HAL.h** utilize the **atomic.h** file.

or lastly remove ProUI from `EXTENSIBLE_UI`

### Bug Timeline

just noticed, but probably when ProUI became under the `EXTENSIBLE_UI` umbrella

### Expected behavior

_No response_

### Actual behavior

Error when compiling

### Steps to Reproduce

1. Load HC32 configs (provided)
2. `HC32F460C_aquila_101` in **platformio.ini**
3. Compile

### Version of Marlin Firmware

bugfix-2.1.x

### Printer model

_No response_

### Electronics

_No response_

### LCD/Controller

_No response_

### Other add-ons

_No response_

### Bed Leveling

None

### Your Slicer

None

### Host Software

None

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

[HC32configs.zip](https://github.com/user-attachments/files/16260704/HC32configs.zip)
> [!NOTE]
> in `Configuration.h` you may need to edit `#define HEATER_0_MAXTEMP 315` to be `275`

--- a/Marlin/src/HAL/HC32/HAL.h
+++ b/Marlin/src/HAL/HC32/HAL.h
@@ -114,22 +114,19 @@
 // Misc. Functions
 //
 #ifndef analogInputToDigitalPin
-#define analogInputToDigitalPin(p) pin_t(p)
+  #define analogInputToDigitalPin(p) pin_t(p)
 #endif
 
-#define CRITICAL_SECTION_START        \
-  uint32_t primask = __get_PRIMASK(); \
-  (void)__iCliRetVal()
+#define CRITICAL_SECTION_START()        \
+  const bool irqon = !__get_PRIMASK();  \
+  __disable_irq();                      \
+  __DSB();
+#define CRITICAL_SECTION_END()          \
+  __DSB();                              \
+  if (irqon) __enable_irq();
 
-#define CRITICAL_SECTION_END \
-  if (!primask)              \
-  (void)__iSeiRetVal()
-
-// Disable interrupts
-#define cli() noInterrupts()
-
-// Enable interrupts
-#define sei() interrupts()
+#define cli() __disable_irq()
+#define sei() __enable_irq()
 
 // bss_end alias
 #define __bss_end __bss_end__

